src/ui/public/visualize/visualize_legend.js

```diff
@@ -68,7 +68,7 @@ uiModules.get('kibana')
 
       $scope.showLegendCount = legendData => {
         if (!$scope.vis.params.addLegendCount) return false;
-        if (!$scope.vis.type.isLegendCountSupported) return false;
+        if (!$scope.vis.type.params.isLegendCountSupported) return false;
         return  $scope.getLegendCount(legendData);
```

Comment:
Based on the name of this function (`showLegendCount`) and its usage, I expect it to return a boolean value. So consider changing the final return statement to `return !!$scope.getLegendCount(legendData);` or `return Boolean($scope.getLegendCount(legendData));`